ToDoList-Redux/Views/CheckmarkCellView.swift

```diff
@@ -19,22 +19,32 @@ struct CheckmarkCellView: View {
             Image(systemName: viewModel.state.imageName)
                 .resizable()
                 .frame(width: 20, height: 20)
-                .onTapGesture { viewModel.dispatch(.toggle(viewModel.state.id))}
+                .onTapGesture { viewModel.dispatch(.toggle) }
             Text(viewModel.state.title)
         }
     }
 }
 
+#if DEBUG
 struct CheckmarkCellView_Previews: PreviewProvider {
+    static let stateMock = AppState.mock
+    static let mockStore = ObservableViewModel<AppAction, AppState>.mock(
+        state: stateMock,
+        action: { action, _, state in
+            state = Reducer.app.reduce(action, state)
+        }
+    )
     static var previews: some View {
         Group {
-            CheckmarkCellView(viewModel: .mock(state: .mock,
-                                               action: { action, _, state in
-                                                state = Reducer.cell.reduce(action, state)
-                                               }
-            )
+            CheckmarkCellView(
+                viewModel:
+                    CheckmarkCellView.viewModel(
+                        store: mockStore.projection(action: AppAction.task, state: \AppState.tasks),
```

Comment:
You can even use these previews for Snapshot Testing.

Reply:
Would love to hear more 🧐

Reply:
https://www.youtube.com/watch?v=tk0HzScvW2M

She uses this library: https://github.com/pointfreeco/swift-snapshot-testing
While you're there, you may wanna check that Point-free also has a redux framework that shares some similarities with SwiftRex, but with some different concepts and completely different implementation. The benefit is that they probably have a larger community ready to give support. But there are some significant differences and you may like more one or another. Since you're comparing libraries, I think you'd like to know.